torch/csrc/lazy/core/ir.cpp

```diff
@@ -59,32 +60,28 @@ bool Node::enableDynamicShape() {
   return enabled || FLAGS_ltc_enable_dynamic_shapes;
 }
 
-Node::Node(OpKind op, size_t num_outputs, hash_t node_hash, std::function<hash_t(bool)> dag_hash_fn)
+Node::Node(OpKind op, size_t num_outputs, hash_t node_hash)
     : op_(op),
       num_outputs_(num_outputs),
       node_hash_(node_hash),
-      dag_hash_without_sizes_(dag_hash_fn(false)),
-      dag_hash_with_sizes_(dag_hash_fn(true)),
       metadata_(GetMetaDataIfDebugging()) {}
 
 Node::Node(OpKind op, size_t num_outputs, std::function<hash_t(bool)> node_hash_fn)
     : op_(op),
       num_outputs_(num_outputs),
-      node_hash_(node_hash_fn(!enableDynamicShape())),
       dag_hash_without_sizes_(node_hash_fn(false)),
       dag_hash_with_sizes_(node_hash_fn(true)),
+      node_hash_(enableDynamicShape() ? dag_hash_without_sizes_ : dag_hash_with_sizes_),
       metadata_(GetMetaDataIfDebugging()) {}
 
 
 Node::Node(OpKind op, OpList operands, std::vector<Shape>&& shapes,
            size_t num_outputs, hash_t hash_seed)
-    : Node(op, num_outputs,
-           // TODO(WHC) this is inefficient (having to compute node_hash twice
-           // since I can't call hash() yet) so probably move dag_hash
-           // initialization to a separate function?
-           /* node_hash */ HashCombine(op.hash(), hash_seed),
-           /* dag_hash */
-           [&](bool bakeInSizes) { return OperandHashes(operands, HashCombine(op.hash(), hash_seed), bakeInSizes); }) {
+    : Node(op, num_outputs, /* node_hash */ HashCombine(op.hash(), hash_seed)) {
+  // Initialize dag hashes
+  dag_hash_without_sizes_ = OperandHashes(operands, node_hash_, false);
```

Comment:
i think i would not propose to change this further in this PR, but while i am looking at it, I want to discuss what it takes to remove the _without_sizes_ and _with_sizes hashes from Node entirely.

If we just go back to Node having a single 'dag_hash' field, we could just initialize it with `node_hash_fn(enableDynamicShapes())`.

It would break the shape cache implementation in TS backend.  The TS backend wants to just look up a node's pre-existing hash in the cache.  If we're not in dynamic shapes mode this would work.  If we are, it fails.  Some random ideas...
 - ~let Ts backend compute the 'hash_with_sizes' just in time for the cache usage~
 - ~have the cache designed differently, so it can compare actual input shapes before using a result~
 - move hash_with_sizes field to TsNode class so at least Node/XlaNode doesn't have to compute it, and TsNode can also avoid computing it in static shapes mode

Reply:
I'll revert the changes for now then